### ISSUE TITLE
Nuxt-link navigation does not work with prefix_and_default strategy

### ISSUE THREAD

**Issue by Rantti** [nuxt-modules/i18n#1977]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.3.3`
- Nitro Version:    `2.3.2`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `srcDir`, `modules`, `i18n`, `dev`, `debug`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.10`
- Build Modules:    `-`
------------------------------

package version:

- "@nuxtjs/i18n": "8.0.0-beta.10"

### Reproduction

Minimal repro can be found from here:

https://stackblitz.com/edit/github-ptyzfd?file=nuxt.config.ts

### Describe the bug

This might be related to https://github.com/nuxt-modules/i18n/issues/1864 

Using `prefix_and_default` strategy seems to "disable" the nuxt-links on pages. The `href` is defined correctly on HTML, but navigation is still prevented upon clicking. 

Looking at the i18n plugin, there is a [routeMiddleWare that detects the redirect upon navigation and defines the final target path.](https://github.com/nuxt-modules/i18n/blob/next/src/runtime/plugins/i18n.ts#L438).

[Since the strategy is `prefix_and_default` and we are currently on default locale, the if condition for deciding if a redirect should be done will pass always.](https://github.com/nuxt-modules/i18n/blob/next/src/runtime/utils.ts#L308)

[The routepath is evaluated from either resolving current route for current locale, or then the target path for current locale.](https://github.com/nuxt-modules/i18n/blob/next/src/runtime/utils.ts#L314)

To me it seems, that the redirect-detection should be happening upon arriving to the target route path. Now since the routepath is evaluated with `context.$switchLocalePath(targetLocale) || context.$localePath(fullPath, targetLocale)`, `context.$switchLocalePath(targetLocale)` always gets resolved and is prioritized over `context.$localePath(fullPath, targetLocale)`.

If I switch these two evaluations to  `context.$localePath(fullPath, targetLocale) || context.$switchLocalePath(targetLocale) `, the path becomes resolved correctly. 

I'm only picking this up upon encountering this issue, so I'm not completely familiar with the whole module. Do I understand the intended functionality correctly? Or is there some additional configuration that I should add for the `prefix_and_default` strategy to work? I also understand that simply switching the evaluation on that particular line can probably cause issues elsewhere on the project. I'm happy to help with resolving this issue if this is not working as intended.

For logs, there is included what gets logged to the console, when on starting page, and clicking the link pointing towards `/page2`.

### Logs

```shell
locale-changing middleware 
Object { fullPath: "/page2", path: "/page2", query: {}, hash: "", name: "page2___en___default", params: {}, matched: (1) […], meta: Proxy, redirectedFrom: undefined, href: "/page2" }
Object { fullPath: "/", hash: "", query: {}, name: "index___en___default", path: "/", params: {}, matched: (1) […], meta: Proxy, redirectedFrom: undefined, href: "/" }
i18n.mjs:335
detectLocale: initialLocale - en i18n.utils.mjs:146
detectLocale: detectBrowserLanguage (browserLocale, stat, reason, from) - <empty string> false unknown unknown i18n.utils.mjs:148
detectLocale: finaleLocale first (finaleLocale, strategy) - <empty string> prefix_and_default i18n.utils.mjs:159
detectLocale: finaleLocale second (finaleLocale, detectBrowserLanguage) - en false i18n.utils.mjs:171
detectLocale: finalLocale last (finalLocale, defaultLocale) - en en i18n.utils.mjs:179
detectLocale: finalLocale - en i18n.utils.mjs:183
detect locale en i18n.mjs:348
localeSetup false i18n.mjs:350
setLocale: new ->  en  old ->  en  initial ->  false i18n.utils.mjs:102
detectRedirect: targetLocale ->  en i18n.utils.mjs:188
detectRedirect: route ->  
Object { fullPath: "/page2", path: "/page2", query: {}, hash: "", name: "page2___en___default", params: {}, matched: (1) […], meta: Proxy, redirectedFrom: undefined, href: "/page2" }
i18n.utils.mjs:189
detectRedirect: calculate routePath ->  / /page2 i18n.utils.mjs:198
redirectPath on locale-changing middleware / i18n.mjs:363
navigate options  301 null false false i18n.utils.mjs:236
navigate isSSG false i18n.utils.mjs:237
```

**Comment by alexgrozav** [comment]
Got the same. Spent a whole day on debugging. Something like this should never ever fail silently. Can we also log the error that causes this in the future?

**Comment by hahatulk** [comment]
Same

**Comment by ineshbose** [comment]
Hi, thanks for opening an issue on the repository and sharing your problem!

It seems to work on edge (https://github.com/nuxt-modules/i18n/issues/1864#issuecomment-1436019506) - we're working to release a tagged release soon - you can also downgrade to beta.9 if required.

In your provided reproduction, I do see that `localePath` isn't being used/passed into `nuxt-link`. Moreover, the non-`localePath` paths being passed are relative URLs while you should also use `localePath` there with preferably name of the routes rather than paths - keep in mind that routing is based on Vue Router, so referring to those docs can also be helpful.

This is how I would do it:

```html
<script setup>
const { t } = useI18n()
const localePath = useLocalePath()

const helperFunc = (name) => localePath({name})
</script>

<template>
  <h2>{{ $t('hello') }} root</h2>
  <nuxt-link :to="localePath('page2')">Page 2</nuxt-link>
  and
  <nuxt-link :to="localePath({ name: 'page3' })">Page 3</nuxt-link>
  <br />

  <nuxt-link :to="helperFunc('page2')">Page 2, using localepath</nuxt-link>
</template>
```

We're discussing the option to natively extend `NuxtLink` (and other similar utilities like `navigateTo`, etc) to make use of `localePath` (but this needs to be done carefully to avoid confusion for developers and any unintentional behavior).

**Comment by hahatulk** [comment]
> ```
> const localePath = useLocalePath()
> ```

Thanks for answer. Problem remains even with routing by name

**Comment by ineshbose** [comment]
Were you able to switch to edge?

https://github.com/nuxt-modules/i18n#install-edge-version

**Comment by hahatulk** [comment]
> Were you able to switch to edge?
> 
> https://github.com/nuxt-modules/i18n#install-edge-version

Ive already rollbacked to beta-9. Nvm :)

**Comment by Lord-Y** [comment]
Same pb with version [8.0.0-beta.12-28090693.c8406ad](https://www.npmjs.com/package/@nuxtjs/i18n-edge/v/8.0.0-beta.12-28090693.c8406ad) and Nuxt `3.5.2`

**Comment by a1s** [comment]
The bug is still there in @nuxtjs/i18n-edge version 8.0.0-beta.12-28122654.89ddb67

Here's an example: https://stackblitz.com/edit/node-iguqcm?file=nuxt.config.ts

When non-default locale is selected, the navigation works as expected.  But when the locale is set to _en_, `navigateTo()` always displays the index page.

**Comment by a1s** [comment]
I believe the problem is with this line: https://github.com/nuxt-modules/i18n/blob/e66999fb63eb1d4e074eb8a71be2fec5c8083584/src/runtime/utils.ts#L310

`context.$switchLocalePath(targetLocale)` takes precedence and `fullPath` is never used.

**Comment by kazupon** [comment]
fixed by #2249 
so, you can install edge channel version.
